generic_config_updater/services_validator.py

```diff
@@ -17,13 +17,42 @@ def set_verbose(verbose=False):
 
 def _service_restart(svc_name):
     rc = os.system(f"systemctl restart {svc_name}")
-    logger.log(logger.LOG_PRIORITY_NOTICE,
-            f"Restarted {svc_name}", print_to_console)
+    if rc != 0:
```

Comment:
>rc [](http://example.com/codeflow?start=7&length=2)

Is the `rc` a stable code when too many restarts happen? If yes, just check `if rc == <code>:`

Reply:
rc == or != 0 is stable comparison for success/failure

Reply:
Sorry, I mean if there is a specific exit code for "too many restarts" failure? if yes, we can check that code specifically.

Reply:
There are no specific error code. 1 is a generic error code, which is most commonly we see. But this could change in future. Moreover if there is any inherent service related error, the monitor catches it and leads to ICM.

In our case, for any failure, we try our best, before giving up.

ref: [link](https://refspecs.linuxfoundation.org/LSB_5.0.0/LSB-Core-generic/LSB-Core-generic.html#INISCRPTACT)

"In case of an error while processing any init-script action except for status, the init script shall print an error message and exit with a non-zero status code:"

"1  generic or unspecified error (current practice)"